integrations/gradle-plugin/src/main/groovy/org/robolectric/gradleplugin/GradlePlugin.groovy

```diff
@@ -0,0 +1,92 @@
+package org.robolectric.gradleplugin
+
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Dependency
+import org.gradle.api.artifacts.DependencyResolutionListener
+import org.gradle.api.artifacts.ResolvableDependencies
+
+class GradlePlugin implements Plugin<Project> {
+    public static final String downloadTaskName = "robolectricDownloadAndroidSdks"
+
+    static class Config {
+        Object sdks
+    }
+
+    @Override
+    public void apply(Project project) {
+        Config config = new Config()
+        project.extensions.add("robolectric", config)
+
+        provideSdks(project)
+        addRobolectricDependencies(project)
+        enableIncludeAndroidResources(project)
+    }
+
+    private addRobolectricDependencies(Project project) {
+        String robolectricVersion = loadResourceFile("robolectric-version.txt")
+        def selfTestClassPath = System.getenv("gradle-robolectric-plugin.classpath")
+
+        project.getGradle().addListener(new DependencyResolutionListener() {
+            @Override
+            void beforeResolve(ResolvableDependencies resolvableDependencies) {
+                def configuration = project.getConfigurations().getByName("testImplementation")
+
+                Dependency dependency
+                if (selfTestClassPath != null) {
+                    def parts = selfTestClassPath.split(":")
+                    dependency = project.getDependencies().create(project.files(*parts))
+                } else {
+                    dependency = project.getDependencies().create("org.robolectric:robolectric:$robolectricVersion")
+                }
+                configuration.getDependencies().add(dependency)
+
+                project.getGradle().removeListener(this)
+            }
+
+            @Override
+            void afterResolve(ResolvableDependencies resolvableDependencies) {
+            }
+        })
+    }
+
+    private static enableIncludeAndroidResources(Project project) {
+        project.afterEvaluate { p ->
+            Object androidExt = project.getExtensions().findByName("android")
```

Comment:
Nit: If you want to be a bit more type-safe you could also ask for `project.getExtensions.findByType(BaseExtension.class)` - this will return you the correct instance of either  `ApplicationExtension`, `LibraryExtension`, `TestExtension` or `DynamicFeatureExtension` (all named `android`), based on what Android plugin was actually applied.